File: solution/1190. Reverse Substrings Between Each Pair of Parentheses/1190.cpp
class Solution {
public:
    string reverseParentheses(string s) {
        stack<int> L;
        for (int i = 0; i < s.size(); ++i) {
            if (s[i] == '(') {
                L.push(i);
            } else if (s[i] == ')') {
                reverse(s.begin() + L.top(), s.begin() + i);
                L.pop();
            }
        }
        
        string res;
        for (char& c : s) {
            if (c == '(' || c == ')') continue;
            res += c;
        }
        return res;
    }
};



// (u(love)i)
// (ueovli

File: solution/1190. Reverse Substrings Between Each Pair of Parentheses/1190.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        L = []
        s = list(s)  # 將字符串轉換為列表，因為 Python 字符串是不可變的

        for i in range(len(s)):
            if s[i] == '(':
                L.append(i)
            elif s[i] == ')':
                s[L[-1]+1:i] = s[L[-1]+1:i][::-1]
                L.pop()

        return ''.join(c for c in s if c not in '()')
